students/arun_nalla/session03/Lesson3_Mailroom.py

```diff
@@ -16,14 +16,13 @@ def thanks_email(seq):
                             "We highly appreciate your kindness. Till date you have donate a total of ${:.2f}.\n\n"
                             "With Regards\n\nXYZ\n".format(name_input, donation_amount, sum(seq[i][1])))
                     return seq
-            for i, v in enumerate(seq):
-                if name_input != seq[i][0]:
-                    donation_amount = float(input('Enter donation amount: '))
-                    seq.append((name_input, [donation_amount]))
-                    print("Dear {},\nThis message is regarding the recent donation of ${:.2f} you made to ZYX charity.\n"
+            else:
```

Comment:
Good use of `else`, I think, but you don't even need this `else` at all, because the only case when the execution continues after the `for` loop is when there's no match. Therefore, `else` is already implied when the execution reaches here, so that's why I think `else` is not even needed here. Just FYI, no need to change.